ATB/BusinessLogic/Services/Implementations/BookingService.cs

```diff
@@ -0,0 +1,125 @@
+using BusinessLogic.DTOs;
+using BusinessLogic.Services.Interfaces;
+using DataAccess.Enums;
+using DataAccess.Models;
+using DataAccess.Repositories.Interfaces;
+using DataAccess.SearchCriteria;
+
+namespace BusinessLogic.Services.Implementations;
+
+public class BookingService(
+    IBookingRepository bookingRepository,
+    IFlightRepository flightRepository,
+    IPassengerRepository passengerRepository)
+    : IBookingService
+{
+    public async Task<BookingDto?> GetByIdAsync(Guid id)
+    {
+        var booking = await bookingRepository.GetByIdAsync(id);
+        return booking is null ? null : new BookingDto(booking);
+    }
+
+    public async Task BookFlight(Guid flightId, Guid passengerId, FlightClass flightClass)
+    {
+        var flight = await flightRepository.GetByIdAsync(flightId);
+        if (flight == null)
+        {
+            throw new ArgumentException($"Flight with id {flightId} not found");
+        }
+
+        // if (!(await IsClassAvailableToBook(flight, flightClass)))
+        // {
+        //     throw new ArgumentException($"Class {flightClass} not available for flight with id {flightId}");
+        // }
```

Comment:
if unneeded ,please delete

Reply:
Nit : is committed code needed here ?

Reply:
Yes,  I uncommented it in my last commits